enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,467 @@
+        - --registry=txt
+        - --txt-owner-id=externalDNS-operator
+        - --log-level=debug
+        # Needs to be unique per container
+        - --metrics-address=:7979
+        - --zone-id-filter=/hostedzone/myzone
+        env:
+          # Base64 decode from cloud-credentials secret generated by cloud credentials request
+          - name: AWS_ACCESS_KEY_ID
+            value: abc
+          - name: AWS_SECRET_ACCESS_KEY
+            value: xyz
+      - name: external-dns-public
+        image: k8s.gcr.io/external-dns/external-dns:v0.8.0
+        args:
+        - --source=openshift-route
+        - --provider=aws
+        - --policy=upsert-only # would prevent ExternalDNS from deleting any records, omit to enable full synchronization
+        # can only have one zone at a time if domains are overlapping over zones
+        - --aws-zone-type=public
+        - --registry=txt
+        - --txt-owner-id=my-identifier
+        - --log-level=debug
+        # Needs to be unique per container
+        - --metrics-address=:7980
+        # Filter from aws dev account
+        - --zone-id-filter=/hostedzone/myzone
+        env:
+          # Base64 decode from cloud-credentials secret generated by cloud credentials request
+          - name: AWS_ACCESS_KEY_ID
+            value: abc
+          - name: AWS_SECRET_ACCESS_KEY
+            value: xyz
+```
+
+Note the use of multiple ExternalDNS containers within the same pod int he above deployment. Metrics listening ports would have
+to be configured on a per-container basis to avoid collisions.
+
+Metrics will be exposed via an insecure channel by default.
+On OpenShift, externalDNS instances can be deployed with a reverse proxy sidecar, such as kube-rbac-proxy, to provide a secure means of metrics scraping.
+
+### Risks and Mitigations
+
+Cluster administrators may not want to give users the ability to create arbitrary DNS records in an external provider when they grant users permission to create routes.
+ExternalDNS could observe a sub-field of a route resource when determining whether or not to publish records to the upstream provider.
+Cluster administrators could restrict a users ability to set this route sub-field, using a mechanism similar to that for route custom hosts.
+
+Similarly, cluster administrators may want to restrict the ability for users to specify the ExternalDNS
+[DNS target override annotation](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/openshift.md#prepare-router_canonical_hostname-in-defaultrouter-deployment).
+on route resources, since the override annotation could give users the ability to create DNS records for alternative domains.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. If an OpenShift route is exposed by multiple Ingress Controllers, should ExternalDNS publish multiple records?
+1. Can more than one ExternalDNS instance (ie. one for a public zone and one for a private zone) be ran using a single deployment?
```

Comment:
This does seem to deviate from upstream that uses a deployment-per-ext-dns instance.